### DISCUSSION TITLE
Sustainability Includes?

### DISCUSSION THREAD

**Discussion by ericproegler** [m3tomlins/performacology#4] [General]
For discussing the goals of this project, I wrote up some ideas about Sustainability. It could include the following:

1. CPU cost - there is a cost for every activity; inefficiency wastes money and time (response means users' time too - a precious commodity)
2. Reliability - crashes/restarts mean more time spent babysitting systems, and increases TCO. Systems that are expensive to operate are less sustainable
3. xxxx - the label for the emotional/psychic cost for developers and maintainers from unplanned downtime, pressure to fix (#hugops), frustration, and toil from systems that are not resilient and anti-fragile
4. Environmental impact - wasted cycles create CO2. That's unacceptable, even in a world with digital tulips (Grifto) and stochastic parrots  (LLMs)

**Comment by m3tomlins** [comment]
Not go completely wild on this tangent: what if we also considered "sustainable" to humans in ways beyond ecology? Perhaps, working 60 hours a week? Cortisol levels? Mental and emotional well-being? Perhaps a practice could be considered to have high sustainability if the effort and outcomes improve human experience and well-being?

**Reply by hohenner** [reply]
How about sustainability in the dev process, “good practices” that drive easier load on devs, CI/CD, zero downtime deploys, …

**Comment by hohenner** [comment]
speaking of our sustainability channel just posted https://www.linkedin.com/pulse/slash-your-codes-carbon-footprint-github/

**Comment by m3tomlins** [comment]
Nice - thank you for sharing that!

Reduce your carbon footprint via "actually being honest about the fact that you ignore test results and push to prod, so why spend compute resources actually running the tests in the first place if you're just gonna ignore them"

:)

**Comment by ericproegler** [comment]
Wasteful tests and pipeline activities are wasted carbon, time, opportunity, slower speeds eroding dev efficiency - it's easy to get back to a broad sustainability stance that includes happiness and satisfaction. 

I work in a Developer Experience org these days. We're thinking about cognitive load, unplanned work, toil, etc - the things that make the work less enjoyable. Some interesting stuff here about Developer fulfillment: https://dl.acm.org/doi/10.1145/3595878

**Comment by ericproegler** [comment]
Preparing to lecture some SDETs:
219 (7%) of cases never passed. Why are we running them?
23,220 test failures last month. Did we review every failure?
We executed 423,063 tests. Was the cost and CO2 worth it?

**Comment by ericproegler** [comment]
Sustainability Engineering Pattern: Linting that saves time and cycles by finding syntax issues without having to run the code to find the problem
